### PR TITLE
Add Avatar Endpoint

### PR DESCRIPTION
**Pull request by johngriebel**
feature/issue #195 - Add the endpoint, model changes, migration, unit tests.

Signed-off-by: John Griebel <jgriebel@3blades.io>

### CODE REVIEW

**Comment by johngriebel** [comment]
@roachmd - Note that in production OpenResty will have to be configured to serve `MEDIA_URL`, as Django doesn't do this when `DEBUG` is set to `False`.

File: users/models.py [hunk 1]
```diff
@@ -13,9 +13,14 @@ class User(AbstractUser):
...
 
 
+def user_directory_path(instance, filename):
+    return "{username}/{filename}".format(username=instance.user.username,
+                                          filename=filename)
+
+
 class UserProfile(models.Model):
     user = models.OneToOneField(User, primary_key=True, related_name='profile')
-    avatar_url = models.CharField(max_length=100, blank=True, null=True)
```
Comment (jkp85) [inline users/models.py]: Why removing avatar_url. Issue is about adding upload feature. What if user would want to use url? @jgwerner What do you think?

Comment (johngriebel) [inline users/models.py]: Avatar url can still be accessed via `user.profile.avatar.url`.

Comment (jkp85) [inline users/models.py]: That's not what I meant. What if I want to use [gravatar](https://gravatar.com) or S3? Initial idea was to allow users use urls that they already have.

Comment (raygesualdo) [inline users/models.py]: I think this is a good first pass. I think we could wait for URLs until we have SSO OAuth set up with 3rd parties (GitHub, etc.).

Comment (johngriebel) [inline users/models.py]: I see what @jkp85 is saying. I had forgotten about S3, etc. There's no reason to _remove_ `avatar_url` if it already exists and we will use it in the future anyhow. I'll make this change.

**Pull request merged**